src/main/java/de/tum/in/www1/artemis/domain/SubmissionPatch.java

```diff
@@ -0,0 +1,36 @@
+package de.tum.in.www1.artemis.domain;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import de.tum.in.www1.artemis.domain.modeling.ModelingSubmissionPatch;
+import de.tum.in.www1.artemis.domain.participation.Participation;
+
+/**
+ * A SubmissionPatch is a DTO that is used to update a submission.
+ * It contains the participation and the patch that should be applied to the submission,
+ * in the format of a JSON patch (RFC 6902).
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "submissionExerciseType")
+// Annotation necessary to distinguish between concrete implementations of Submission when deserializing from JSON
+// @formatter:off
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = ModelingSubmissionPatch.class, name = "modeling"),
+})
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public abstract class SubmissionPatch extends DomainObject {
```

Comment:
does it really make sense to put this into the domain package which is used for database objects?
This is more a DTO in my opinion and should be a Java Record

Reply:
Also I think inheritance is not needed here! We only need a ModelingSubmissionPatch for now!

Reply:
I was trying to be both future-proof and conforming to the convention of Submission DTO, not restricting the implementation to modelling submissions except when it is needed (which was in the websocket handler). This would also make it more likely that future developers follow the same convention when implementing patching for text or programming exercises.

But I can also remove it.

Reply:
You are correct, I mistakenly used an entity for mapping patches. Additionally, a singular record suffices for modelling submission patches. I removed the entity and added a single `SubmissionPatch` record in websocket dtos.